### PR TITLE
Thread multicast DFU support

### PR DESCRIPTION
**Pull request by rlubos**
This PR adds support for Thread multicast DFU, that will be added in future Thread SDK release.

### CODE REVIEW

**Comment by rlubos** [comment]
Thank you for the review. I've applied appropriate fixes and removed some obsoletecode , that we've considered as no longer needed - mostly regarding the `diag` option.

**Comment by mrodem** [comment]
Thanks :+1: I see that there is also a conflict in `nordicsemi/__main__.py` that needs to be resolved. Also, the CLA must be signed before the PR can be merged.

File: nordicsemi/version.py [hunk 1]
```diff
@@ -37,4 +37,4 @@
 
 """ Version definition for nrfutil. """
 
-NRFUTIL_VERSION = "3.3.2"
+NRFUTIL_VERSION = "3.4.0"
```
Comment (bihanssen) [inline nordicsemi/version.py]: @rlubos, you asked me if the version should be bumped as part of the PR, which I confirmed. However, in hindsight, I think it is better to leave out the version change here, and we'll instead deal with setting the correct version when we do the next release. Sorry for the mixup.

Comment (rlubos) [reply nordicsemi/version.py]: Ok, I've removed that commit.

**Comment by rlubos** [comment]
Ok, so I've signed the CLA (I wasn't aware that Noridc employees have to do this as well) and removed the commit that bumbed up the version.

Is there any change though that updated version will be released any time soon? We wanted to point out in our SDK's documentation to that specific version, and we're releasing soon.

**Comment by IvanSanchez** [comment]
This is failing on one of my Linux machines with the following error:

```
$ nrfutil dfu thread -pkg /tmp/Thread_DFU-test/app_dfu_package.zip --port /dev/ttyACM0  -f
pkg /tmp/Thread_DFU-test/app_dfu_package.zip --port /dev/ttyACM0  -f
Address not specified. Using ff03::1 (all Thread nodes)
Using connectivity board at serial port: /dev/ttyACM0
Board already flashed with connectivity firmware.
Opening serial to /dev/ttyACM0 @ 115200
2017-11-10 16:55:45,473 Failed to set property PROP_IPv6_ICMP_PING_OFFLOAD
Traceback (most recent call last):
  File "/home/ivan/devel/pc-nrfutil/nordicsemi/__main__.py", line 918, in thread
    transport.open()
  File "/usr/local/lib/python2.7/dist-packages/nordicsemi/thread/tncp.py", line 211, in open
    if (not self._attach_to_network()):
  File "/usr/local/lib/python2.7/dist-packages/nordicsemi/thread/tncp.py", line 102, in _attach_to_network
    self._set_property(*props)
  File "/usr/local/lib/python2.7/dist-packages/nordicsemi/thread/tncp.py", line 87, in _set_property
    raise Exception("Failed to set property {}".format(self.__class__._propid_to_str(propid)))
Exception: Failed to set property PROP_IPv6_ICMP_PING_OFFLOAD
```

@rlubos I'd rather make this work locally at least once before merging. Any clues?

**Comment by wbober** [comment]
@IvanSanchez 
The issue you see occurs sometimes after board has been flashed with ncp connectivity firmware. Usually a single hard reset helps.

**Comment by IvanSanchez** [comment]
@wbober No luck. It's either `Failed to set property PROP_IPv6_ICMP_PING_OFFLOAD` after running with `-f`, or `Failed to reset NCP. Please flash connectvity firmware.` after a reset or power-cycle.

**Comment by rlubos** [comment]
@IvanSanchez I do not have solution, just some feedback. 
I've tried nrfutil on Linux for the first time (Ubuntu 16.04 on VirtualBox) and experienced the same issue you described. But in my case, power cycle helped, and since then I've not experienced such a failure (iv'e tried like 10 times). I'm running nrfutil with `-f` option all the time.

**Comment by IvanSanchez** [comment]
Apparently this runs, but I haven't been able to complete a multicast DFU, and also I'm not totally happy with some of the error messages and behaviour.

I'll merge this, but I'll also create a few issues about those.

**Pull request merged**